cms/static/cms/js/modules/cms.structureboard.js

```diff
@@ -1295,16 +1298,108 @@ class StructureBoard {
+     * and if the were present befor. If they weren't present before the will be downloaded
+     * and executed. If ther script also has the class 'cms-trigger-load-events' the
+     * 'load' and 'DOMContentLoaded' events will be triggered
+     *
+     * @param {JQuery} new_scripts  JQuery selector of the scripts for the new body content
+     * @param {JQuery} old_scripts  JQuery selector of the scripts for the old body content
+     */
+    addJsScriptsNeededForRender(new_scripts, old_scripts) {
+        const scriptSrcList = [];
+        let classListCollection = [];
+        const that = this;
+
+        new_scripts.each(function() {
+            let script_exists = false;
+            let new_script = $(this);
+
+            if (new_script.hasClass('cms-execute-js-to-render')) {
+                $(old_scripts).each(function() {
+                    let old_script = $(this);
+
+                    if (new_script.prop('outerHTML') === old_script.prop('outerHTML')) {
+                        script_exists = true;
+                        return false;
+                    }
+                });
+                if (!script_exists) {
+                    let classList = new_script.attr('class').split(' ');
+
+                    classListCollection = classListCollection.concat(classList);
+                    if (typeof new_script.prop('src') === 'string' && new_script.prop('src') !== '') {
+                        scriptSrcList.push(new_script.prop('src'));
+                    } else {
+                        let js_file = document.createElement('script');
+
+                        js_file.textContent = new_script.prop('textContent') || '';
+                        js_file.type = 'text/javascript';
+                        document.body.appendChild(js_file);
+                    }
+                }
+            }
+        });
+        if (scriptSrcList.length === 0) {
+            that.triggerLoadEventsByClass(classListCollection);
+        } else {
+            that.getMultiScripts(scriptSrcList).done(function() {
+                that.triggerLoadEventsByClass(classListCollection);
+            });
+        }
+    }
+
+    /**
+     * Downloads and executes all scripts given by `scriptSrcList` and returns a
+     * `Promise` when all are done. This is to prevent multiple triggering of
+     * load events by `triggerLoadEventsByClass` in `addJsScriptsNeededForRender`.
+     * Original solution from https://stackoverflow.com/a/11803418
+     *
+     * @param {array} scriptSrcList   Array of script sources
+     *
+     * @returns {Promise}
+     */
+    getMultiScripts(scriptSrcList) {
+        let _arr = $.map(scriptSrcList, function(scr) {
+            return $.getScript(scr);
+        });
+
+        // eslint-disable-next-line new-cap
+        _arr.push($.Deferred(function(deferred) {
+            $(deferred.resolve);
+        }));
+
+        return $.when.apply($, _arr);
```

Comment:
maybe this whole method could be boiled down `return Promise.all(scriptSrcList.map(s => $.getScript(s))`?

it's ok to use es6 features since the source is transpiled anyway

Reply:
I will try it out

Reply:
But I don't think so, since this isn't a 'proper Promise' but a [jQuery implementation](https://api.jquery.com/jQuery.getScript/) with different methods.

Reply:
It works, I will cherry pick the commit from my test branch, after all CI tests succeeded and push it here.